components/dashboard/src/app/AppLoading.tsx

```diff
@@ -5,8 +5,19 @@
  */
 
 import { FunctionComponent } from "react";
+import { Delayed } from "../components/Delayed";
+import { Heading3, Subheading } from "../components/typography/headings";
+import gitpodIcon from "../icons/gitpod.svg";
 
-// TODO: Would be great to show a loading UI if it's been loading for awhile
 export const AppLoading: FunctionComponent = () => {
-    return <></>;
+    return (
+        // Wait 3 seconds before showing the loading screen to avoid flashing it too quickly
+        <Delayed wait={3000}>
+            <div className="flex flex-col justify-center items-center w-full h-screen space-y-4">
+                <img src={gitpodIcon} alt="Gitpod's logo" className={"h-16 flex-shrink-0 animate-bounce"} />
+                <Heading3>Just getting a few more things ready</Heading3>
+                <Subheading>hang in there...</Subheading>
```

Comment:
**suggestion**: If we're keeping the loading message, I'd go with something less verbose and drop the heading above, as this page does not need it.